third_party/blink/renderer/modules/ml/v2/operand.h

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef THIRD_PARTY_BLINK_RENDERER_MODULES_ML_OPERAND_H_
+#define THIRD_PARTY_BLINK_RENDERER_MODULES_ML_OPERAND_H_
+
+#include "base/logging.h"
+#include "services/ml/public/mojom/model.mojom-blink.h"
+#include "third_party/blink/public/platform/web_string.h"
+#include "third_party/blink/renderer/core/typed_arrays/dom_array_buffer_view.h"
+#include "third_party/blink/renderer/modules/ml/v2/operand_descriptor.h"
+#include "third_party/blink/renderer/platform/bindings/script_wrappable.h"
+#include "third_party/blink/renderer/platform/heap/heap_allocator.h"
+
+namespace blink {
+
+int32_t StringToUnsignedLong(const WebString& operand_type);
+
+void AddFuseOperand(ml::mojom::blink::ModelInfoPtr& model_info);
+void AddUnspecifiedOperand(ml::mojom::blink::ModelInfoPtr& model_info);
+void AddOperand(const OperandDescriptor* descriptor,
+                ml::mojom::blink::ModelInfoPtr& model_info);
+void SetOperandValue(
+    uint32_t index,
+    DOMArrayBufferView* data,
+    HeapHashMap<WTF::String, Member<DOMArrayBufferView>>& buffer_views,
+    ml::mojom::blink::ModelInfoPtr& model_info);
+void AddOperation(int32_t type,
+                  const Vector<uint32_t>& inputs,
+                  const Vector<uint32_t>& outputs,
+                  ml::mojom::blink::ModelInfoPtr& model_info);
+void IdentifyInputsAndOutputs(const Vector<uint32_t>& inputs,
+                              const Vector<uint32_t>& outputs,
+                              ml::mojom::blink::ModelInfoPtr& model_info);
```

Comment:
I don't want those subclass of `Operand` to depend on NNModel header file, how about adding a util file to host those common function?

Reply:
> I don't want those subclass of `Operand` to depend on NNModel header file

Do you have any specific reason for that?

Actually, you can wrap the `ModelInfoPtr` into `NNModel` that is an implementation details.

Reply:
> Do you have any specific reason for that?

I prefer the dependence as simple as possible, do you mean move those common function to `NNModel.h ` in namedspeace?

> you can wrap the ModelInfoPtr into NNModel

It can't resolve this dependence?

Reply:
I mean to make these methods as members of `NNModel`

Reply:
It also makes sence, i will move those common function.